ci/terraform/sns.tf

```
@@ -0,0 +1,125 @@
+#This is  SNS topic created in us-east-1 region for cloudwatch Alarm for Cloudfront
+
+resource "aws_sns_topic" "slack_events" {
+  provider                         = aws.cloudfront
+  count                            = var.cloudfront_auth_frontend_enabled ? 1 : 0
+  name                             = "${var.environment}-cloudfront-alerts"
+  lambda_failure_feedback_role_arn = aws_iam_role.sns_logging_iam_role[0].arn
+
+  tags = local.default_tags
+}
+
+data "aws_iam_policy_document" "sns_topic_policy" {
+  version  = "2012-10-17"
+  provider = aws.cloudfront
+  count    = var.cloudfront_auth_frontend_enabled ? 1 : 0
+
+  statement {
+    actions = [
+      "SNS:Subscribe",
+      "SNS:SetTopicAttributes",
+      "SNS:RemovePermission",
+      "SNS:Receive",
+      "SNS:Publish",
+      "SNS:ListSubscriptionsByTopic",
+      "SNS:GetTopicAttributes",
+      "SNS:DeleteTopic",
+      "SNS:AddPermission",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
```

```suggestion
 principals {
      type        = "Service"
      identifiers = ["cloudwatch.amazonaws.com"]
    }

    condition {
      test     = "StringEquals"
      values   = [data.aws_caller_identity.current.account_id]
      variable = "aws:SourceAccount"
    }
```
Does this work instead? I think this topic policy gives anyone in the world permissions?